components/taxonomy/src/values.rs

```diff
@@ -1601,53 +1783,23 @@ impl PartialOrd for Value {
         use std::cmp::Ordering::*;
         match (self, other) {
             (&Unit, &Unit) => Some(Equal),
-            (&Unit, _) => None,
-
             (&OnOff(ref a), &OnOff(ref b)) => a.partial_cmp(b),
-            (&OnOff(_), _) => None,
-
             (&OpenClosed(ref a), &OpenClosed(ref b)) => a.partial_cmp(b),
-            (&OpenClosed(_), _) => None,
-
             (&DoorLocked(ref a), &DoorLocked(ref b)) => a.partial_cmp(b),
-            (&DoorLocked(_), _) => None,
-
+            (&Secure(ref a), &Secure(ref b)) => a.partial_cmp(b),
             (&Duration(ref a), &Duration(ref b)) => a.partial_cmp(b),
-            (&Duration(_), _) => None,
-
             (&TimeStamp(ref a), &TimeStamp(ref b)) => a.partial_cmp(b),
-            (&TimeStamp(_), _) => None,
-
             (&Temperature(ref a), &Temperature(ref b)) => a.partial_cmp(b),
-            (&Temperature(_), _) => None,
-
             (&Color(ref a), &Color(ref b)) => a.partial_cmp(b),
-            (&Color(_), _) => None,
-
             (&ExtBool(ref a), &ExtBool(ref b)) => a.partial_cmp(b),
-            (&ExtBool(_), _) => None,
-
             (&ExtNumeric(ref a), &ExtNumeric(ref b)) => a.partial_cmp(b),
-            (&ExtNumeric(_), _) => None,
-
             (&String(ref a), &String(ref b)) => a.partial_cmp(b),
-            (&String(_), _) => None,
-
             (&Json(ref a), &Json(ref b)) => a.partial_cmp(b),
-            (&Json(_), _) => None,
-
             (&ThinkerbellRule(ref a), &ThinkerbellRule(ref b)) => a.name.partial_cmp(&b.name),
-            (&ThinkerbellRule(_), _) => None,
-
             (&WebPushNotify(ref a), &WebPushNotify(ref b)) => a.resource.partial_cmp(&b.resource),
-            (&WebPushNotify(_), _) => None,
-
             (&Binary(self::Binary {mimetype: ref a_mimetype, data: ref a_data}),
              &Binary(self::Binary {mimetype: ref b_mimetype, data: ref b_data})) if a_mimetype == b_mimetype => a_data.partial_cmp(b_data),
-            (&Binary(_), _) => None,
-
-            (&Range(_), &Range(_)) => None,
-            (&Range(_), _) => None,
+            _ => None
```

Comment:
Please don't do that. I realize that the existing code is more verbose, but it lets the compiler spot forgotten cases. Adding a catch-all clause breaks this nice property.